### ISSUE TITLE
awkward.argsort does not return correct types for empty lists

### ISSUE THREAD

**Issue by lgray** [scikit-hep/awkward#802]
ak.argsort does not return an index-type in the case that the input list array is empty!

```python
>>> import awkward as ak
>>> ak.__version__
'1.2.0'
>>> x = ak.Array([[1.1, 2.2], [3.3, 3.1]])
>>> x
<Array [[1.1, 2.2], [3.3, 3.1]] type='2 * var * float64'>
>>> x[x > 5]
<Array [[], []] type='2 * var * float64'>
>>> ak.argsort(x[x > 5])
<Array [[], []] type='2 * var * float64'>
>>> ak.argsort(x)
<Array [[0, 1], [1, 0]] type='2 * var * int64'>
```

**Comment by ianna** [comment]
@lgray - thanks for spotting it! The https://github.com/scikit-hep/awkward-1.0/pull/803 should fix it.

**Comment by ianna** [comment]
fixed via #803
